plugins/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3Repository.java

```diff
@@ -260,6 +265,12 @@ public S3Repository(RepositoryName name, RepositorySettings repositorySettings,
         this.chunkSize = getValue(repositorySettings, Repository.CHUNK_SIZE_SETTING, Repositories.CHUNK_SIZE_SETTING);
         this.compress = getValue(repositorySettings, Repository.COMPRESS_SETTING, Repositories.COMPRESS_SETTING);
 
+        // We make sure that chunkSize is bigger or equal than/to bufferSize
+        if (this.chunkSize.getMb() < bufferSize.getMb()) {
```

Comment:
True. At the beginning I thought that `bytes()` would not work for `tb` or `pb` but I figured it later and forgot to change this part of the code.
Thanks for catching this!